### ISSUE TITLE
Very noisy pictures with com.android.camera2

### ISSUE THREAD

**Issue by magcks** [WeAreFairphone/android_device_fairphone_FP3#16]
Hey there,

I really don't know if this is a issue related to this project. So please excuse if this is the wrong place for this issue.

I finally received my FP3 and directly flashed this lineage port onto it and noticed that images are really noisy when captured with com.android.camera2 (see attachment) compared with my old nexus 5x. Obviously only in low-brightness environments.

Is this related to non-existing noise-reduction magic, some issues with the os or is just my hardware broken?

Max
![IMG_20200626_193349](https://user-images.githubusercontent.com/2052401/85887122-7e267780-b7e7-11ea-875c-41a90d707e6c.jpg)

**Comment by DaAngus** [comment]
I had a similar experience and switched to Opencamera, Settings, misc: Camera2 API 
That fixed it for me

**Comment by magcks** [comment]
The package name of com.android.camera2 indicates it also uses the Camera2 API or is this just misleading?

I also tried both APIs with Opencamera but always hat that strange noise artifacts.

**Comment by mohe2015** [comment]
Can you check the ISO settings? The image looks like the ISO is way too high.

**Comment by magcks** [comment]
I checked them. ISO values seems reasonable. I noticed that that strange chromatic noise does not exist that heavy on the stock image (I still don't know if these is an issue with my hardware, because I cannot compare against another FP3). Both images were caputured with opencam with same ISO and EXP. It looks like the stock rom does some smoothing in front of the camera API or so. Espacially if you look at the carton on the carpet, you will realise the differences. I real-world scenarios that looks very disturbing in some cases.

Lineage:
![opencam_lineage](https://user-images.githubusercontent.com/2052401/86464237-5ecfa300-bd2f-11ea-9f8c-79c934d27df7.jpg)
Stock:
![opencam_stock](https://user-images.githubusercontent.com/2052401/86464242-60996680-bd2f-11ea-9877-625e040665fc.jpg)

**Comment by dk1978** [comment]
Hey, sorry for being late on this thing. To be honest - I don't see much of a difference between "Lineage" and "Stock". My experience with the camera has been ok so far.

**Comment by mstaz** [comment]
After using camera several time now I also can see that noise.

> Stock:
> ![opencam_stock](https://user-images.githubusercontent.com/2052401/86464242-60996680-bd2f-11ea-9877-625e040665fc.jpg)

This also doesn't look very nice to me. Maybe a general issue with FP3?

**Comment by magcks** [comment]
@dk1978 yes, that's true. They are almost equal. Probably the differences are just subjectively...

My thoughts are that they rely more on the noise reduction stuff of google camera. I'm currently discussing this issue with the FP-Support. The thing I notice is that open camera images look extremely bad espacially if pixels are close to image boundaries.

For now, I also think that this is a general issue with all FP3 or at least just some FP3. Thus, I will close this issue for now.

**Comment by CNugteren** [comment]
I'm also seeing a lot of noise, especially when it gets darker. But even when light is slightly reduced, e.g. early in the evening, it is noticeable without zooming in. I made a few test on an FP3 with LineageOS with the original camera app (similar results with OpenCamera btw) as well as with [PXv4.5_GoogleCamera_7.2.014](https://www.celsoazevedo.com/files/android/google-camera/f/changelog1330/) on LineageOS, and for comparison also on an FP3 with the original ROM. I 'solved' the noise issue thus by installing this GoogleCamera port.

Here is a resized version of some images at reasonable lighting conditions:

* LineageOS standard camera app (very noisy):
![FP3_lineage_small](https://user-images.githubusercontent.com/1162108/92411007-7586e180-f146-11ea-8ee5-0eca9d9d5e25.jpg)
* PXv4.5_GoogleCamera (not noisy): 
![FP3_GcamPX_small](https://user-images.githubusercontent.com/1162108/92411004-7455b480-f146-11ea-9bc8-ba5ec65c41bc.jpg)
* Original FP3 ROM (not noisy):
![FP3_original_small](https://user-images.githubusercontent.com/1162108/92411009-76b80e80-f146-11ea-92a9-b63c5ab03822.jpg)

And here is a close-up of the full-resolution:

* LineageOS standard camera app (very noisy):
![FP3_lineage_cropped](https://user-images.githubusercontent.com/1162108/92411069-aff07e80-f146-11ea-9e18-e17768d43dc7.jpg)
* PXv4.5_GoogleCamera (not noisy): 
![FP3_GcamPX_cropped](https://user-images.githubusercontent.com/1162108/92411068-af57e800-f146-11ea-8dd0-134d9b16e432.jpg)
* Original FP3 ROM (not noisy):
![FP3_original_cropped](https://user-images.githubusercontent.com/1162108/92411070-b121ab80-f146-11ea-99a6-fa796587bdad.jpg)

**Comment by magcks** [comment]
Thanks for your photos.

In the mean time I got a little further into the problems you can observe from your "LineageOS standard camera app" photo. I think these are hot pixels that remains static during multiple shots. You can observe this by take photos of complete darkness (i.e. put something in front of your cam) and save the resulting image as a DNG file (opencamera).

Probably, both camera apps do either better noise reduction or dark frame subtractions.
